From: Maxime Ripard <maxime.ripard@free-electrons.com>
Date: Tue, 27 Mar 2018 14:19:42 +0200
Subject: [PATCH 22/31] Add an option to disable IDLE

IDLE is an IDE embedded into python, written using Tk, so it doesn't make
much sense to have it into our build.

Signed-off-by: Maxime Ripard <maxime.ripard@free-electrons.com>
[ Andrey Smirnov: ported to Python 3.6 ]
Signed-off-by: Andrey Smirnov <andrew.smirnov@gmail.com>
---
 Makefile.pre.in | 7 ++++++-
 configure.ac    | 7 +++++++
 setup.py        | 4 +++-
 3 files changed, 16 insertions(+), 2 deletions(-)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index 1ff2c9b..d51d64e 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -1194,7 +1194,9 @@ bininstall: altbininstall
 	-rm -f $(DESTDIR)$(LIBPC)/python3.pc
 	(cd $(DESTDIR)$(LIBPC); $(LN) -s python-$(VERSION).pc python3.pc)
 	-rm -f $(DESTDIR)$(BINDIR)/idle3
+ifeq (@IDLE@,yes)
 	(cd $(DESTDIR)$(BINDIR); $(LN) -s idle$(VERSION) idle3)
+endif
 	-rm -f $(DESTDIR)$(BINDIR)/pydoc3
 ifeq (@PYDOC@,yes)
 	(cd $(DESTDIR)$(BINDIR); $(LN) -s pydoc$(VERSION) pydoc3)
@@ -1239,7 +1241,6 @@ LIBSUBDIRS=	site-packages \
 		html json http dbm xmlrpc \
 		logging csv wsgiref urllib \
 		ctypes ctypes/macholib \
-		idlelib idlelib/Icons \
 		distutils distutils/command \
 		importlib \
 		turtledemo \
@@ -1309,6 +1310,10 @@ ifeq (@EXPAT@,yes)
 LIBSUBDIRS += $(XMLLIBSUBDIRS)
 endif
 
+ifeq (@IDLE@,yes)
+LIBSUBDIRS += idlelib idlelib/Icons
+endif
+
 ifeq (@LIB2TO3@,yes)
 LIBSUBDIRS += lib2to3 lib2to3/fixes lib2to3/pgen2
 TESTSUBDIRS += lib2to3/tests			\
diff --git a/configure.ac b/configure.ac
index 158fcf7..c15a785 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3198,6 +3198,13 @@ AC_ARG_ENABLE(lib2to3,
 	[ LIB2TO3="${enableval}" ], [ LIB2TO3=yes ])
 AC_MSG_RESULT($LIB2TO3)
 
+AC_SUBST(IDLE)
+AC_MSG_CHECKING(for --disable-idle3)
+AC_ARG_ENABLE(idle3,
+	AS_HELP_STRING([--disable-idle3], [disable idle3 IDE]),
+	[ IDLE="${enableval}" ], [ IDLE=yes ])
+AC_MSG_RESULT($IDLE)
+
 # Check for enable-ipv6
 AH_TEMPLATE(ENABLE_IPV6, [Define if --enable-ipv6 is specified])
 AC_MSG_CHECKING([if --enable-ipv6 is specified])
diff --git a/setup.py b/setup.py
index 59dc442..625ecca 100644
--- a/setup.py
+++ b/setup.py
@@ -2324,11 +2324,13 @@ def main():
     import warnings
     warnings.filterwarnings("ignore",category=DeprecationWarning)
 
-    scripts = ['Tools/scripts/idle3', 'Tools/scripts/pyvenv']
+    scripts = ['Tools/scripts/pyvenv']
     if not '--disable-pydoc' in sysconfig.get_config_var("CONFIG_ARGS"):
         scripts += [ 'Tools/scripts/pydoc3' ]
     if not '--disable-lib2to3' in sysconfig.get_config_var("CONFIG_ARGS"):
         scripts += [ 'Tools/scripts/2to3' ]
+    if not '--disable-idle3' in sysconfig.get_config_var("CONFIG_ARGS"):
+        scripts += [ 'Tools/scripts/idle3' ]
 
     setup(# PyPI Metadata (PEP 301)
           name = "Python",
